animation_nodes/nodes/material/set_object_materials.py

```diff
@@ -0,0 +1,35 @@
+import bpy
+from ... base_types import AnimationNode, VectorizedSocket
+
+class SetObjectMaterialsNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SetObjectMaterialsNode"
+    bl_label = "Set Object Materials"
+
+    useList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+        self.newInput(VectorizedSocket("Material", "useList",
+            ("Material", "material"),
+            ("Materials", "materials")), defaultDrawType = "PROPERTY_ONLY")
+
+        self.newOutput("Object", "Object", "object")
+
+    def getExecutionFunctionName(self):
+        if self.useList:
+            return "executeList"
+        else:
+            return "executeSingle"
+
+    def executeSingle(self, object, material):
+        if object == None or not hasattr(object.data, "materials"): return object
```

Comment:
```suggestion
        if object is None or not hasattr(object.data, "materials"): return object
```

Usually checking if something is `None` is done using `is` instead of `==`.